.circleci/config.yml

```diff
@@ -1,56 +1,378 @@
+
+  test-node8-mysql: &mysql-template
+    executor: mysql
+    working_directory: ~/typeorm/node8/
+    steps:
+      - run-db-tests:
+          database: mysql
+
+  test-node8-oracle: &oracle-template
+    executor: oracle
+    working_directory: ~/typeorm/node8/
+    steps:
+      - run-db-tests:
+          database: oracle
+
+  test-node8-postgres: &postgres-template
+    executor: postgres
+    working_directory: ~/typeorm/node8/
+    steps:
+      - run-db-tests:
+          database: postgres
+
+  test-node8-sqlite: &sqlite-template
+    executor: sqlite
+    working_directory: ~/typeorm/node8/
+    steps:
+      - run: npm install sqlite3
+      - run-db-tests:
+          database: sqlite
+
+  test-node8-sqljs: &sqljs-template
+    executor: sqljs
+    working_directory: ~/typeorm/node8/
+    steps:
+      - run-db-tests:
+          database: sqljs
+
+  test-node10:
+    <<: *test-template
+    working_directory: ~/typeorm/node10/
+    docker:
+      - image: circleci/node:10.16.3
+    steps:
+      - prepare-for-tests:
+          node_folder: node8
+      - persist_to_workspace:
+          root: ~/typeorm/
+          paths:
+            - ./node10/
+
+  test-node10-mariadb:
+    <<: *mariadb-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-mongodb:
+    <<: *mongodb-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-mssql:
+    <<: *mssql-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-mysql:
+    <<: *mysql-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-oracle:
+    <<: *oracle-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-postgres:
+    <<: *postgres-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-sqlite:
+    <<: *sqlite-template
+    working_directory: ~/typeorm/node10/
+
+  test-node10-sqljs:
+    <<: *sqljs-template
+    working_directory: ~/typeorm/node10/
+
+  test-node12:
+    <<: *test-template
+    working_directory: ~/typeorm/node12/
+    docker:
+      - image: circleci/node:12.10.0
+    steps:
+      - prepare-for-tests:
+          node_folder: node8
+      - persist_to_workspace:
+          root: ~/typeorm/
+          paths:
+            - ./node12
+
+  test-node12-mariadb:
+    <<: *mariadb-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-mongodb:
+    <<: *mongodb-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-mssql:
+    <<: *mssql-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-mysql:
+    <<: *mysql-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-oracle:
+    <<: *oracle-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-postgres:
+    <<: *postgres-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-sqlite:
+    <<: *sqlite-template
+    working_directory: ~/typeorm/node12/
+
+  test-node12-sqljs:
+    <<: *sqljs-template
+    working_directory: ~/typeorm/node12/
+
+workflows:
```

Comment:
I've started leveraging CircleCI workflows in this config which allow the tests to run in parallel. This is easiest to visualize in the CircleCI dashboard, but essentially:

* `checkout_code` runs to get the code from GH, the copy that code for each version of node to test
* `test_node8`, `test_node10` and `test_node12` run which runs `npm install` for their respective versions of node
* `test_node*_*` runs for each version of node and each DB to test

I'm definitely not happy with how verbose this config is, but as far as I can tell, CircleCI workflows don't allow for parameters or any other way of specifying a matrix of jobs to run for each value. While the config is definitely overly verbose, it doesn't seem too bad since we don't add new DBs that often.